### PR TITLE
Add Create Table As Select test for SQL Server

### PR DESCRIPTION
**Pull request by sanjay990**

### CODE REVIEW

File: presto-product-tests/src/main/java/com/facebook/presto/tests/sqlserver/Select.java [hunk 1]
```diff
@@ -141,4 +142,22 @@ public void testAllDatatypes()
...
+                "CREATE TABLE %s AS SELECT * FROM %s", CREATE_TABLE_AS_SELECT, NATION_TABLE_NAME);
+        onPresto().executeQuery(sql);
+
+        sql = format(
+                "SELECT n_nationkey, n_name, n_regionkey, n_comment FROM %s",
+                CREATE_TABLE_AS_SELECT);
+        QueryResult queryResult = onPresto()
+                .executeQuery(sql);
+
+        assertThat(queryResult).matches(PRESTO_NATION_RESULT);
```
Comment (maciejgrzybek) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/sqlserver/Select.java]: Drop the table, you have just created, when the test finished.

**Review by maciejgrzybek — CHANGES_REQUESTED** [review]

File: presto-product-tests/src/main/java/com/facebook/presto/tests/sqlserver/Select.java (hunk 1 above)
Comment (sanjay990) [reply presto-product-tests/src/main/java/com/facebook/presto/tests/sqlserver/Select.java]: https://github.com/Teradata/presto/pull/515/commits/e75f3a5c7f5ea7bd58167a9d4a3c1c55fc3b3d10

File: presto-product-tests/src/main/java/com/facebook/presto/tests/sqlserver/Select.java [hunk 2]
```diff
@@ -141,4 +142,27 @@ public void testAllDatatypes()
...
+
+        onPresto().executeQuery(format("DROP TABLE IF EXISTS %s", CREATE_TABLE_AS_SELECT));
+        sql = format(
+                "CREATE TABLE %s AS SELECT * FROM %s", CREATE_TABLE_AS_SELECT, NATION_TABLE_NAME);
```
Comment (rschlussel-zz) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/sqlserver/Select.java]: I don't think you mean to create it again too.

File: presto-product-tests/src/main/java/com/facebook/presto/tests/sqlserver/Select.java [hunk 3]
```diff
@@ -159,5 +159,6 @@ public void testCreateTableAsSelect()
                 .executeQuery(sql);
 
         assertThat(queryResult).matches(PRESTO_NATION_RESULT);
+        onPresto().executeQuery(format("DROP TABLE IF EXISTS %s", CREATE_TABLE_AS_SELECT));
```
Comment (maciejgrzybek) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/sqlserver/Select.java]: This won't be triggered if previous queries fail. You need some sort of final-block but without hiding the reason for an actual failure. Have a look how we have this done in `AbstractTestHiveClient` (method `dropTable`). Or, similar approach in `AlterTableTests`, with `dropTestTables` method.

**Review by maciejgrzybek — CHANGES_REQUESTED** [review]

File: presto-product-tests/src/main/java/com/facebook/presto/tests/sqlserver/Select.java [hunk 4]
```diff
@@ -141,4 +157,21 @@ public void testAllDatatypes()
...
+
+    @Test(groups = {SQL_SERVER, PROFILE_SPECIFIC_TESTS})
+    public void testCreateTableAsSelect()
+            throws SQLException
+    {
+        String sql = format(
+                "CREATE TABLE %s AS SELECT * FROM %s", CREATE_TABLE_AS_SELECT, NATION_TABLE_NAME);
+        onPresto().executeQuery(sql);
+
+        sql = format(
```
Comment (maciejgrzybek) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/sqlserver/Select.java:171]: I believe you should not use Presto to check results.
It's like using a test subject for the test itself.

How hard would it be to introduce `onSqlServer().executeQuery(...)`?

File: presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestGrantRevoke.java [hunk 5]
```diff
@@ -35,7 +35,7 @@
 import static java.lang.String.format;
 
 public class TestGrantRevoke
-    extends ProductTest
+        extends ProductTest
```
Comment (maciejgrzybek) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestGrantRevoke.java]: Is the formatting off here?

File: presto-product-tests/src/main/java/com/facebook/presto/tests/sqlserver/Select.java [hunk 6]
```diff
@@ -60,9 +64,23 @@ public Requirement getRequirements(Configuration configuration)
                 immutableTable(SQLSERVER_ALL_TYPES));
     }
 
+    private static final String CTAS = "create_table_as_select";
```
Comment (maciejgrzybek) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/sqlserver/Select.java]: `CTAS_TABLE_NAME`

**Review by maciejgrzybek — APPROVED** [review]

**Comment by maciejgrzybek** [comment]
Looks good. Two nits mentioned.

Make sure Travis passes, though!

**Pull request closed without merging**